tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,499 @@
+cJF+r+l6F3IcRPi4z/eaEKG7uhAS59TjH2tAAAAFQC0I9kL3oceMT1O
+44WPe6NZ8w8CMwAAAIABGm2Yg8nGFZbo/W8njuM79w0W2P1NBVNWzBH
+WQqVbr4i1bWTSSc9X+itQUpeF6zAUDsUoprhNise2NLrMYCLFo9JxhE
+iYAcEJ/YbKEnjtJzaAmQNpyh3rCWuOcGPTevjAZIkl+zEc+/N7tCW1e
+uDYm6IXZ8LEQyTUQUdU4pZ2OgAAAIABk1ZA3+TiCMaoAafNVUZ7zwqk
+888yVOgsJ7HGGDGRMo5ytr2SUJB7QWsLX6Un/Zbu32nXsAqtqagxd6F
+Ies98TSekMh/hAv9uK92mEsXSINXOeIMKRedqOyPgk5IEOsFpxAUO4T
+xpYToeuM8HRemecxw2eIFHnax+mQqCsi7FgQ== core@valid2",
+                ),
+                (
+                    "valid3",
+                    "
+sk-ssh-ed25519@openssh.com AAAAGnNrLXNzaC1lZDI1NTE5QG9w
+ZW5zc2guY29tAAAAIEX/dQ0v4127bEo8eeG1EV0ApO2lWbSnN6RWusn
+/NjqIAAAABHNzaDo= demos@siril",
+                ),
+                (
+                    "bad",
+                    "
+    ssh-bad this-not-a-key core@bad",
+                ),
+            ].iter().cloned().collect();
+
+        static ref FINGERPRINTS: HashMap<&'static str, &'static str> =
+            [
+                (
+                    "valid1",
+                    "SHA256:yZ+o48h6quk9c+JVgJ/Zq4S5u4LUk6TSpneHKkmM9KY",
+                ),
+                (
+                    "valid2",
+                    "SHA256:RP5k1AybZ1kollIAnpUavr1v1nfZ0yloKvI46AMDPkM",
+                ),
+                (
+                    "valid3",
+                    "SHA256:U8IKRkIHed6vFMTflwweA3HhIf2DWgZ8EFTm9fgwOUk",
+                ),
+            ].iter().cloned().collect();
+    }
+
+    #[test]
+    fn setup_tests() {
+        for (name, text) in &*TEST_KEYS {
+            _ = fs::create_dir_all(SSH_DIR);
+            let pub_path: PathBuf = PathBuf::from(format!("{}/{}.pub", SSH_DIR, name));
+            let mut pubfile = File::create(pub_path).expect("unable to create a file");
+            let _ = pubfile.write_all(text.as_bytes());
+        }
+    }
+
+    fn assert_has_keys(keys: Vec<&str>) {
+        let authkeyspath: PathBuf = PathBuf::from(format!("{}/authorized_keys", SSH_DIR));
+        let authkeysfile = File::open(authkeyspath);
+        assert!(authkeysfile.is_ok());
+        let mut authkeystext = String::new();
+        authkeysfile.unwrap().read_to_string(&mut authkeystext).expect("unable to read a file");
+        assert!(authkeystext.starts_with("# auto-generated"));
+
+        for key in &keys {
+            assert!(authkeystext.contains(key));
+        }
+
+        for (key, _) in &*TEST_KEYS {
+            if keys.contains(&key) {
+                continue;
+            }
+            assert!(!authkeystext.contains(key));
+        }
+    }
+
+    fn run_script(args: Vec<&str>) -> Result<Popen, PopenError> {
+        let username = match users::get_current_username() {
+            Some(name) => name,
+            None => panic!("unable to get a current username"),
+        };
+        Popen::create(
+            [&[SCRIPT_PATH, "-u", username.into_string().unwrap().as_str(), "--ssh-dir", SSH_DIR], args.as_slice()].concat().as_slice(),
```

Comment:
Instead of calling the compiled binary which will only work once `cargo build` ran, you could rather call a high-level function here that does the work of the main function.

Reply:
Reimplemented based on high-level functions.